### PR TITLE
sampleHeight fixes and cleanup

### PR DESCRIPTION
**Pull request by lilleyse**
Fixes https://github.com/AnalyticalGraphicsInc/cesium/issues/7120 and https://github.com/AnalyticalGraphicsInc/cesium/pull/7129

Otherwise most of this is organizational changes from https://github.com/AnalyticalGraphicsInc/cesium/pull/7115 that will make that PR easier to review later.

@bagnell can you review?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @lilleyse!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Core/Ray.js [hunk 1]
```diff
@@ -38,6 +38,25 @@ define([
...
+     * @returns {Ray} The modified result parameter or a new Ray instance if one was not provided. (Returns undefined if ray is undefined)
+     */
+    Ray.clone = function(ray, result) {
+        if (!defined(ray)) {
+            return undefined;
+        }
+        if (!defined(result)) {
+            return new Ray(ray.origin, ray.direction);
+        }
+        result.origin = ray.origin;
```
Comment (bagnell) [inline Source/Core/Ray.js]: Here and below should clone `ray.origin` and `ray.direction`.

File: Source/Scene/Cesium3DTile.js [hunk 2]
```diff
@@ -607,6 +609,63 @@ define([
...
+            if (tileset.dynamicScreenSpaceError) {
+                var density = tileset._dynamicScreenSpaceErrorComputedDensity;
+                var factor = tileset.dynamicScreenSpaceErrorFactor;
+                var dynamicError = CesiumMath.fog(distance, density) * factor;
+                error -= dynamicError;
+            }
+        }
+        return error;
+    };
+    /**
```
Comment (bagnell) [inline Source/Scene/Cesium3DTile.js]: Whitespace.

File: Source/Scene/Cesium3DTile.js [hunk 3]
```diff
@@ -607,6 +609,63 @@ define([
 
     var scratchJulianDate = new JulianDate();
 
+    /**
+     * Get the tile's screen space error.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.getScreenSpaceError = function(frameState, useParentGeometricError) {
```
Comment (bagnell) [inline Source/Scene/Cesium3DTile.js]: Is `useParentGeometricError` ever `false`? Is that in a future PR?

Comment (lilleyse) [reply Source/Scene/Cesium3DTile.js]: It's undefined when called from `updateVisibility`. I forgot to change that to false which should make it more explicit.

**Comment by bagnell** [comment]
Those are the only comments. Looks good to me.

**Comment by lilleyse** [comment]
@bagnell Thanks, updated.

**Comment by Soni-pro** [comment]
I have code like,three entities, a gltfModel and with polyline and label on top of the polyline. I have been using sampleHeight(), to find the height of the gltfModel. but it didn't work as expected. Is there any other method to do that. Or what to do in this?

Thank You

**Pull request merged**